packages/workbox-core/utils/assert.mjs

```diff
@@ -1,11 +1,27 @@
+import WorkboxError from '../models/WorkboxError.mjs';
+
 /*
  * This method returns true if the current context is a service worker.
  */
-const isSWEnv = () => {
-  return (typeof ServiceWorkerGlobalScope !== 'undefined' &&
-    self instanceof ServiceWorkerGlobalScope);
+const isSWEnv = (moduleName) => {
+  if (!(typeof ServiceWorkerGlobalScope !== 'undefined' &&
+    self instanceof ServiceWorkerGlobalScope)) {
+      throw new WorkboxError('not-in-sw', {moduleName});
+  }
+};
+
+const isArray = (value, {moduleName, className, funcName, paramName}) => {
```

Comment:
You're destructuring `{moduleName, className, funcName, paramName}` only to restructure it again later. Why not just just an object?

Reply:
Only reason for this is that if you were looking to use this at some point, this shows what you are expected to supply to call this method, compared to:

```javascript
isArray(value, details) => {
 ...
}
```

Shall I add a comment just to explain this?

Reply:
I don't think it's a huge deal either way. For a public method, I'd expect to see a jsdoc comment for the param explaining the available options. If you're going to comment I'd use a jsdoc comment, otherwise I think it's fine as is.

Reply:
This will be all internal (assertions are stripped completely from production builds)